### PR TITLE
Configure CircleCI

### PR DESCRIPTION
**Pull request by croaky**
* Move off Travis to Circle.
* Run the entire script on macOS infrastructure.
* Performance optimization:
  Enforce a Ruby version to skip Ruby install

We're cheating slightly with this implementation,
but seems worth it for speed of build tradeoff:

* Use pre-installed XCode for compilers, etc.
* Use pre-installed Ruby version.

Related:

https://github.com/thoughtbot/laptop/issues/494

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+dependencies:
+  cache_directories:
+    - /usr/local/Homebrew
+  override:
+    - brew install shellcheck
+
+machine:
```
Comment (iwz) [inline circle.yml]: thoughts on using Circle CI 2.0 configuration?

https://circleci.com/docs/2.0/sample-config/

**Comment by croaky** [comment]
It seemed like some of the longest time spent here was on installing Ruby, which I thought would have been covered by the machine config:

```
Configuring Ruby ...
rbenv: version `2.4.1' is not installed (set by /Users/distiller/laptop/.ruby-version)
Downloading ruby-2.4.1.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.4/ruby-2.4.1.tar.bz2
Installing ruby-2.4.1...
ruby-build: use readline from homebrew
Installed ruby-2.4.1 to /Users/distiller/.rbenv/versions/2.4.1
```

I think I'm going to try `echo`ing a `.ruby-version` into the CI box.

File: circle.yml (hunk 1 above)
Comment (croaky) [inline circle.yml]: @iwz I don't have experience yet with Circle 2.0 so was shying away from it for now. Thoughts on what the benefits of it are?

Comment (iwz) [inline circle.yml]: it's containerized with docker, so you can add custom containers easily. probably not important for this repo. not sure how long 1.0 will be around, though

Comment (croaky) [inline circle.yml]: Cool, cool. I had seen Docker as the headline feature but agree maybe better to merge this so @derekprior can use it for his active changes on other PRs and keep our eye on Circle 2.0 as we (I?) get more experience with it.

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+dependencies:
+  cache_directories:
+    - /usr/local/Homebrew
```
Comment (derekprior) [inline circle.yml]: Will this cache the homebrew installs we make when running the script? Or only after the setup script step?

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    - echo '2.4.1' >> /Users/distiller/laptop/.ruby-version
+
+machine:
+  ruby:
+    version: 2.4.1
+  xcode:
+    version: 8.3.3
+
+test:
+  post:
```
Comment (derekprior) [inline circle.yml]: Should this be `override`? Is the current test step doing anything?

File: circle.yml (hunk 2 above)
Comment (croaky) [inline circle.yml]: I think it's only caching the work done in `dependences > override`, e.g. caching `shellcheck`. Maybe worth removing this for clarity.

File: circle.yml (hunk 3 above)
Comment (croaky) [inline circle.yml]: Default test step doesn't do anything, will fail without some config. `override` better, good call.

**Comment by croaky** [comment]
> I think I'm going to try echoing a `.ruby-version` into the CI box.

That did not do the trick. Think I'm going to merge and then see if we can optimize that post-https://github.com/thoughtbot/laptop/pull/502

**Pull request merged**